src/main/java/duke/logic/conversations/Conversation.java

```diff
@@ -0,0 +1,105 @@
+package duke.logic.conversations;
+
+import duke.commons.MessagesPrompt;
+import duke.commons.exceptions.DukeDateTimeParseException;
+import duke.commons.exceptions.DukeException;
+import duke.logic.parsers.ParserTimeUtil;
+import duke.logic.parsers.ParserUtil;
+
+/**
+ * Abstract class representing individual Conversation.
+ */
+public abstract class Conversation {
+    protected String result;
+    protected String prompt;
+    protected int state;
+    private boolean isFinished;
+    protected int attempts;
```

Comment:
you need to say what these protected variables mean in the code. 

see a comment related to `attempts` in a later